libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt

```diff
@@ -82,13 +82,21 @@ class MapboxRouteOverviewButton : ConstraintLayout {
     }
 
     /**
-     * Invoke the function to show optional text associated with the view.
+     * Invoke the function to show default text associated with the view.
      * @param duration for the view to be in the extended mode before it starts to shrink.
      */
     fun showTextAndExtend(duration: Long) {
+        showTextAndExtend(context.getString(R.string.mapbox_route_overview), duration)
```

Comment:
We can have just one function.

fun showTextAndExtend(duration: Long, text: String = context get string(R.string.mapbox_route_overview))

This way you can pass both the arguments or just duration and the function will take the default value.

Same for other classes

Reply:
That won't work for `MapboxSoundButton`, because there will be two optional arguments this way: `text` and `callback`. With the overloads, generated via `@JvmOverloads` annotation, Java users won't be able to provide `callback` without providing some `text` first.

Reply:
I think we can change the API in `MapboxSoundButton`. Earlier `callback` in `MapboxSoundButton` was triggered when the animation was finished. But now, with the change in this [PR](https://github.com/mapbox/mapbox-navigation-android/pull/4710/files?file-filters%5B%5D=.kt&file-filters%5B%5D=.xml) when a user invokes,
```
mapboxSoundButton.muteAmdExtend()
```
this function does the following:
```
mute(callback)
showTextWithAnimation()
```
There is no need of `callback` anymore as we are immediately returning the state of mute/unmute without waiting for the animation to finish.

We should change the API signature as follows
```
@JvmOverloads
fun mute(): Boolean {
    binding.soundButtonIcon.setImageDrawable(muteDrawable)
    return true
}

@JvmOverloads
fun unmute(): Boolean {
    binding.soundButtonIcon.setImageDrawable(inmuteDrawable)
    return false
}
```

After the change we can then change the API's for `MapboxRouteOverViewButton`, `MapboxRecenterButton` and `MapboxSoundButton` as mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/4703#discussion_r689794872)

cc @Zayankovsky @Guardiola31337

Reply:
Do we really need to return `Boolean` here? I see no real value in this. 
Also `callback` in `MapboxSoundButton` was always triggered immediately before starting the animation.

Reply:
Yeah `Boolean` should be returned because since this is a toggle functionality, the end user might want to have a different behavior based on toggle. If you look at [lines](https://github.com/mapbox/mapbox-navigation-android/blob/main/examples/src/main/java/com/mapbox/navigation/examples/core/MapboxVoiceActivity.kt#L165-L175) you will see that volume value is changed based on boolean value returned. Also, asynchronous callback makes sense when we are doing an asynchronous operation. But in this case all `mute/unmute` does is change the image drawable. So we can definitely make is synchronous. Also, it reduces the extra LOC an end user needs to write thus making the API simple to use.

Reply:
But the value being returned is constant, how can it be of any help to the user?
The user doesn't need to check it, because mute functions always return true and unmute functions always return false.

Reply:
As a developer, whatever is happening inside mute/unmute is a black box to me. I would like to know that when I invoke mute/unmute the SDK tells me that following action has been done in some way. I think of this API just like a checkbox which returns true if you click on it otherwise false. In this case as well it's always constant. Hence, instead of calling mute and assuming that it's done, I would like for the function to return me a value. At the moment the code snippet inside mute/unmute is very simple. But in the future if we change the internal logic and need to return true/false then we will break the API. I feel there is no harm in returning true/false starting right now. As a user it's up to you if you want to use the value or not use it and make any assumption. 

However, let's remove the callback as that is definitely not needed anymore.